pkg/app/server.go

```diff
@@ -66,10 +70,20 @@ func (pl promLogger) Log(v ...interface{}) error {
 	return nil
 }
 
-// RunKubeStateMetrics will build and run the kube-state-metrics.
+// RunKubeStateMetricsWrapper runs KSM with context cancellation.
+func RunKubeStateMetricsWrapper(ctx context.Context, opts *options.Options, factories ...customresource.RegistryFactory) error {
+	err := runKubeStateMetrics(ctx, opts, factories...)
+	if ctx.Err() == context.Canceled {
+		klog.Infoln("RESTARTING kube-state-metrics, metrics will be reset")
```

Comment:
```suggestion
		klog.Infoln("Restarting: kube-state-metrics, metrics will be reset")
```

Reply:
Does this mean the container will shut down when the config file is changed?

Reply:
No the container will stay up, only the context will be cancelled, and a new goroutine will be created in place of the older one, which will be destroyed.